### ISSUE TITLE
Alchemy: Automatic gold creation

### ISSUE THREAD

**Issue by mightybyte** [feuerbach/test-framework-golden#6]
It would be nice if goldenVsFile automatically created the golden file with the output file if the gold file doesn't exist.  This gives the user an easy way to recreate gold data when things change during development--simply delete the gold files and re-run the test suite.

**Comment by feuerbach** [comment]
When things change, you should use the `--accept` flag. It is IMO easier and safer than removing golden files.

OTOH, I like this idea, primarily for the case when the test has just been added. Then simply creating the golden file is probably the right thing to do.

A pull request is welcome. It should have an option to turn this off, but the default should probably be "on".

**Comment by feuerbach** [comment]
Also, you should really migrate to tasty-golden. https://groups.google.com/forum/#!search/%22test-framework-golden%22/haskell-cafe/OaJ2rRFfmpE/UsVhDnZYH7IJ

In fact, this very repostiroty is likely to be deleted soon.

**Comment by mightybyte** [comment]
Oh, I didn't know about the --accept flag (it's not mentioned anywhere in the documentation).  I agree that it is much better than removing golden files once you have them.  But the flag doesn't work for me.  When I run `dist/build/test/test --accept`, I get "unrecognized option '--accept'".

On the topic of migrating to tasty, I think that there are still plenty of people using test-framework, so it might not be feasible for people to switch.  Obviously if you don't have the desire to continue maintaining this package that's fine, but it would be nice to have some kind of notice about that in the cabal file so it will be visible on the main hackage page.

**Comment by feuerbach** [comment]
Yeah, the flag only works in tasty-golden. I just remembered that the inability to implement `--accept` with test-framework was one of the reasons I started tasty in the first place.

> On the topic of migrating to tasty, I think that there are still plenty of people using test-framework, so it might not be feasible for people to switch.

That there exist plenty of people using test-framework shouldn't prevent any single of them from migrating, or did you mean something else?

> it would be nice to have some kind of notice about that in the cabal file so it will be visible on the main hackage page that the package is unmaintained

Good point; I somehow forgot to do that. Done.

**Comment by mightybyte** [comment]
> That there exist plenty of people using test-framework shouldn't prevent any single of them from migrating, or did you mean something else?

Oh yeah, my wording wasn't very good there.  I was thinking two thoughts.  First, there are still plenty of people using test-framework and it is still being maintained, so I think it should still be considered as a viable alternative to tasty.  Second, some people may have large amounts of existing tests using test-framework which might make it infeasible for them to convert to tasty.  The point I was driving at is that I don't think you should deprecate test-framework-golden, but rather put a note there that it will be unmaintained until someone else comes along to take up the maintainership.

**Comment by feuerbach** [comment]
For my purposes it *is* deprecated; as in I recommend migrating to tasty, but don't preclude the existing users from installing it.

In case a new maintainer appears, s/he'll take the deprecation warning down, of course.
